discovery-provider/src/tasks/update_track_is_available.py

```diff
@@ -228,7 +229,16 @@ def update_track_is_available(self) -> None:
 
     have_lock = update_lock.acquire(blocking=False)
     if have_lock:
+        metric = None
```

Comment:
this feels like more maint burden to have separate existence checks on `metric` here vs. other jobs https://github.com/AudiusProject/audius-protocol/blob/master/discovery-provider/src/tasks/index_metrics.py#L502

e.g. which method should future reader pick?

can we move the save_time into the try block like the others?

Reply:
if the worker throws, dont we still want metrics on how long it took?

Reply:
ya good point. we should shoot for consistency and make the others have that same behavior too (in future). @joaquincasares . maybe we should just throw a decorator into the mix that auto pulls the name of the job so we don't have to have accidental discrepancies

Reply:
lgtm

Reply:
@vicky-g let's keep the same behavior as the others and only show successful metrics if we aren't going to add more labels.

However, I do like your approach! Thanks for catching that!

Let's please add these changes if we wish to also catch failed tasks:

* Use the labels `("task_name", "success")` on line 237.
* Add a metric.save_time() with the `{success: true}` label after `update_tracks_is_available_status()` is called.
* Add a metric.save_time() with the `{success: false}` label within the `except` block.

/cc @raymondjacobson so you're in loop of how to best deal with successful/failed metrics.

Reply:
gotcha. ya. i think this generally leaves a bit too much design up to the consumer (where we'll end up doing diff things like in this case).

Reply:
@raymondjacobson I gotcha, here are a few examples to help see common patterns: 

* https://www.notion.so/audiusproject/Metrics-Prometheus-Grafana-ae87cebb12af47a087a7eb2c5fd0040f#1a11ddea68e1497884a9f6985f3aab69

But +1 on standardizing across all of our celery tasks. I'm doing that now under #3356 today.

While @vicky-g 's initial PR is still accurate it does deviate a bit from our current pattern. I'd much rather we stick with the same pattern so that way we don't have to remember that the `update_track_is_available_duration_seconds` metric is our only metric that also counts failed durations.

However, I'm all for consumers to add more "sub-work" metrics when they see a good opportunity for them. This PR was a great example!

While not ideal, do please keep @mentioning me on similar PRs until the team is comfortable with how to best design metrics that work for them. Hopefully a few rounds of reviews is all that's needed. :)

Reply:
Resolving for a cleaner demo. :)